app/src/main/java/com/devshish/internship/presentation/ui/MainActivity.kt

```diff
@@ -42,35 +48,82 @@ class MainActivity : AppCompatActivity() {
             )
         )
 
-        val navHostFragment = supportFragmentManager
-            .findFragmentById(R.id.navHostFragment) as NavHostFragment
-        navController = navHostFragment.findNavController()
+        with(binding) {
+            val navHostFragment = supportFragmentManager
+                .findFragmentById(R.id.navHostFragment) as NavHostFragment
+            navController = navHostFragment.findNavController()
+
+            bottomNavView.setupWithNavController(navController)
+
+            navController.addOnDestinationChangedListener { _, destination, _ ->
+                bottomNavView.isVisible = when (destination.id) {
+                    R.id.splashFragment,
+                    R.id.authFragment,
+                    R.id.webFragment,
+                    R.id.playerFragment -> false
+                    else -> true
+                }
+
+                layoutPlayerBar.root.isVisible = when (destination.id) {
+                    R.id.playerFragment -> false
+                    else -> playerViewModel.check()
+                }
```

Comment:
How about remove this and just make visible, and leave TODO here, to later add it in separate task
Because chack like this in fragment is not ok - because for example you can stop your player completely, what will mean that you need to hide whole bar